File: leetcode/easy/js/657.js
const solution = (moves) => {
  let up = 0, down = 0, left = 0, right = 0
  moves.split('').map(move => {
    switch (move) {
      case 'U':
        up++;
        break;
      case 'D':
        down++;
        break;
      case 'L':
        left++;
        break;
      case 'R':
        right++;
        break;
      default:
        return false
    }
  })
  if (
    Math.abs(up) === Math.abs(down) &&
    Math.abs(left) === Math.abs(right)
  ) {
    return true
  }
  return false
}

const input = 'UD'

console.log(solution(input))

File: leetcode/easy/py/657.py
def solution(A):
  up = 0
  down = 0
  left = 0
  right = 0
  movesList = list(A)
  for move in movesList:
    if move == 'U':
      up += 1
    if move == 'D':
      down += 1
    if move == 'L':
      left += 1
    if move == 'R':
      right += 1
    
  if (abs(up) == abs(down) and abs(left) == abs(right)):
    return True
  return False

  
input = 'UD'

print(solution(input))